src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -527,7 +530,7 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
 
                     //our new contact already has an OCR, delete the old one
                     } else {
-                        dmlWrapper.objectsToDelete.add(ocrOldCon);
+                        delete ocrOldCon;
```

Comment:
This method (`manageOtherOCR`) is called by other methods from inside a for loop, so issuing individual dml operations within that for loop could be problematic. Is there a reason you moved away from adding the object to the `dmlWrapper.objectsToDelete` collection?

Reply:
Yes, this is a timing issue with updating OCRs. There was a previous GitHub issue that had a similar timing issue when updating the Primary Contact on an Opportunity when that Contact already had an OCR for that Opportunity.

Reply:
Is there an alternative way to handle this so that it's not executing DML inside a for loop?